src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/ListMutatorFactory.java

```diff
@@ -97,18 +111,95 @@ public void initInPlace(List<T> reference, PseudoRandom prng) {
       }
     }
 
+    private void eraseRandomChunk(List<T> list, PseudoRandom prng, int minSize) {
+      int chunkSize =
+          prng.closedRange(1, Math.max(1, Math.min(list.size() - minSize, list.size() / 2)));
+      int chunkOffset = prng.closedRange(0, list.size() - chunkSize);
+      Iterator<T> iterator = list.listIterator(chunkOffset);
+      int elementsRemoved = 0;
+      while (iterator.hasNext() && elementsRemoved < chunkSize) {
+        iterator.next();
+        iterator.remove();
+        elementsRemoved++;
+      }
+    }
+
+    private void insertRandomChunk(List<T> list, PseudoRandom prng, int maxSize) {
+      // Enforces slower growth for smaller lists and lists that are close to
+      // `maxSize`.
+      int chunkSize = prng.closedRange(1, Math.min(maxSize - list.size(), list.size()));
+      int chunkOffset = prng.indexIn(list);
+      for (int i = 0; i < chunkSize; i++) {
+        T newElement = elementMutator.init(prng);
+        list.add(chunkOffset + i, newElement);
+      }
+    }
+
+    private void changeRandomChunk(List<T> list, PseudoRandom prng) {
+      int chunkOffset = prng.indexIn(list);
+      // Muatate at most 10% of the original list
+      int chunkSize = Math.min(
+          prng.closedRange(1, list.size() - chunkOffset), (int) Math.ceil(list.size() / 10.0));
+      ListIterator<T> iterator = list.listIterator(chunkOffset);
+      int elementsChanged = 0;
+      while (iterator.hasNext() && elementsChanged < chunkSize) {
+        T element = iterator.next();
+        T mutatedElement = elementMutator.mutate(element, prng);
+        iterator.set(mutatedElement);
+        elementsChanged++;
+      }
+    }
+
+    private static enum Action {
+      SHRINK,
+      SHRINK_CHUNK,
+      GROW,
+      GROW_CHUNK,
+      CHANGE,
+      CHANGE_CHUNK,
+    }
+
     @Override
     public void mutateInPlace(List<T> reference, PseudoRandom prng) {
       List<T> list = underlyingMutableList(reference);
       if (list.isEmpty()) {
+        // Early return
         list.add(elementMutator.init(prng));
-      } else if (!prng.trueInOneOutOf(4)) {
-        int i = prng.indexIn(list);
-        list.set(i, elementMutator.mutate(list.get(i), prng));
-      } else if (list.size() < maxSize) {
-        list.add(list.get(list.size() - 1)); // TODO: Create deep copy or init new entry.
-      } else if (list.size() > minSize) {
-        list.remove(list.size() - 1);
+        return;
+      }
+      List<Action> s = new ArrayList<>();
```

Comment:
Building up the list of possible `Action`s could be extracted out into a static method in the enum class, expecting a `Collection`. With that, it could be reused for other collection mutators as well.

Reply:
Extracted into a support class so I can use it for the map improvements in the other PR